src/Client/webpack.config.js

```diff
@@ -1,76 +1,116 @@
+// File adapted from https://github.com/fable-compiler/webpack-config-template
 var path = require("path");
-var webpack = require("webpack");
-var fableUtils = require("fable-utils");
 
 function resolve(filePath) {
-  return path.join(__dirname, filePath)
+    return path.join(__dirname, filePath)
 }
 
-var babelOptions = fableUtils.resolveBabelOptions({
-  presets: [
-    ["env", {
-      "targets": {
-        "browsers": ["last 2 versions"]
-      },
-      "modules": false
-    }]
-  ],
-  plugins: ["transform-runtime"]
-});
-
+var CONFIG = {
+    fsharpEntry:
+        ["whatwg-fetch",
+            "@babel/polyfill",
+            resolve("./Client.fsproj")
+        ],
+    outputDir: resolve("./public"),
+    devServerPort: undefined,
+    devServerProxy: {
+        '/api/*': {
+            target: 'http://localhost:' + (process.env.SUAVE_FABLE_PORT || "8085"),
+            changeOrigin: true
+        }
+    },
+    historyApiFallback: {
+        index: resolve("./index.html")
+    },
+    contentBase: __dirname,
+    // Use babel-preset-env to generate JS compatible with most-used browsers.
+    // More info at https://github.com/babel/babel/blob/master/packages/babel-preset-env/README.md
+    babel: {
+        presets: [
+            ["@babel/preset-env", {
+                "targets": {
+                    "browsers": ["last 2 versions"]
+                },
+                "modules": false,
+                "useBuiltIns": "usage",
+            }]
+        ],
+        plugins: ["@babel/plugin-transform-runtime"]
+    }
+}
 
 var isProduction = process.argv.indexOf("-p") >= 0;
-var port = process.env.SUAVE_FABLE_PORT || "8085";
 console.log("Bundling for " + (isProduction ? "production" : "development") + "...");
 
+var path = require("path");
+var webpack = require("webpack");
+var MinifyPlugin = require("terser-webpack-plugin");
+
+var commonPlugins = [
+];
+
 module.exports = {
-  devtool: "source-map",
-  entry: resolve('./Client.fsproj'),
-  mode: isProduction ? "production" : "development",
-  output: {
-    path: resolve('./public'),
-    publicPath: "/public",
-    filename: "bundle.js"
-  },
-  resolve: {
-    symlinks: false,
-    modules: [ resolve("../../node_modules/")]
-  },
-  devServer: {
-    proxy: {
-      '/api/*': {
-        target: 'http://localhost:' + port,
-        changeOrigin: true
-      }
+    entry: CONFIG.fsharpEntry,
+    // NOTE we add a hash to the output file name in production
+    // to prevent browser caching if code changes
+    output: {
+        path: CONFIG.outputDir,
+        publicPath: "/public",
+        filename: '[name].js'
```

Comment:
Shouldn’t this be either [name][hash].js for production or [name].js for development builds ?

Reply:
The problem is about supporting SSR.

> Also, in general in the different projects, I use the hashed names but here because we are supporting SSR then I am not using the hashed names. We could support hashed names if we generate the `main.[hash].js` and `vendors.[hash].js` first and then update a file on the server to register the generated names.